### DISCUSSION TITLE
Difference between UnoIslands from Sample and Unoislands.Skia.Wpf?

### DISCUSSION THREAD

**Discussion by DevTKSS** [unoplatform/uno#18764] [Q&A]
Hey there,
I am researching a possible way to use UnoIslands in WPF Framework Projects which would use v4.8
In the UnoIslands.Skia.Wpf Namespace I did find a framework struktured Project and in the App.config file is this:
```
﻿<?xml version="1.0" encoding="utf-8" ?>
<configuration>
    <startup> 
        <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.7" />
    </startup>
</configuration>
```
which made me hope, that there might be any chance to use UnoIslands also for applications which requires to use the .Net Framework for the Time until the IT Department do release the higher versions of that complicating Software in the Company.
Maybe someone can tell me something positiv on this. The UnoIslandsSample is using .Net 6 and there is ther actual framework mentioned.

**Comment by jeromelaban** [comment]
Thanks for the question.

.NET framework 4.8 is not supported by Uno anymore in the context of Uno islands. The file you saw is a remnant of the net47 support that was there in Uno 4.11 and got dropped in 5.0. You'll need to upgrade to .NET 8 or later if you want to use the latest versions of Uno.

**Reply by DevTKSS** [reply]
Okay... I did find this repo here https://github.com/PockyBum522/template-inventor-core-addin.git
could you tell if this, which seems to be targeting net6.0-windows would be able to use with uno? I think I did see .Net 6 in the UWP template of yours, but the project structure of the uwp uno templat is other than the regular uno app template. The Interop which is used there must have the same property like the one I have to use, thats my thought. Maybe with this a kind of workaround would be possible to at least use some .net and could easyer update from this to .net 8 if the dependency that causing this trouble for me is updated to .net 8 in some years

**Reply by felixf4xu** [reply]
so, is the info at https://platform.uno/windows7/ still correct?

**Reply by DevTKSS** [reply]
@felixf4xu I would have understood it this way, that a wpf app shell has to be .net like .net and not .net Framework for beeing useable with uno. Also I am referring to this, i watched yesterday: https://dotnet.microsoft.com/en-us/platform/upgrade-assistant
Where the dotnet team is explaining, the version compartiblity quite well. So maybe you should check your framework dependency with that tool.
But uno docs has this .net version article on that also:
https://platform.uno/docs/articles/net-version-support.html
Which i would read as: if you use uno.sdk 4 and earlyer most compartiblity, but uno.check would most likly not allow to stay on v4. Uno.sdk 5+ the dependency hast to be .net standard 2 ar least, what would also mean, it could work because the referenced dotnet article and https://learn.microsoft.com/en-us/dotnet/standard/net-standard would tell also, that .net standard 2 **is** compartible to .net Framework. And that is what is confusing me on @jeromelaban answer, that it is not 😅 so... Maybe the win7 article you linked @felixf4xu is still correct because you are still able to use .net 2 standard in a uwp uno app🤔
I would not know, how to use uno with .net Framework but would also be happy with a just wpf shell which could make internally use of uno. Maybe with setting the imported uno.sdk package nuget version to 4 but that might cause other dependency issues

@jeromelaban for felix quesion context: #18814

**Reply by jeromelaban** [reply]
> so, is the info at https://platform.uno/windows7/ still correct?

Yes, it is, since .NET 8 still works on Windows 7.

> But uno docs has this .net version article on that also: https://platform.uno/docs/articles/net-version-support.html

Indeed, this page is outdated, we'll update it.

**Reply by felixf4xu** [reply]
> > so, is the info at https://platform.uno/windows7/ still correct?
> 
> Yes, it is, since .NET 8 still works on Windows 7.

Thanks, but I'm afraid I don't quite understand your statement.

first of all, I agree to use the word of "**works** on Windows 7", my understanding of it is like:
supported: officially supported by Microsoft, not declared as "end of life".
works: no matter if it's "end of life" or not, it can be installed and run, like people can still run Win7.

So when you say ".NET 8 still works on Windows 7", does it mean .NET 8 runtime (not necessarily SDK) can be installed and run on Win7? I'm not sure of it, because Win7 is not mentioned in .NET8 (https://github.com/dotnet/core/blob/main/release-notes/8.0/supported-os.md) and Win7 is clearly listed as "Out of support" in .NET7 (https://github.com/dotnet/core/blob/main/release-notes/7.0/supported-os.md#out-of-support-os-versions) and .NET6 (https://github.com/dotnet/core/blob/main/release-notes/6.0/supported-os.md#out-of-support-os-versions).

Again, it's written as "Out of **support**", maybe .NET8 still **works** in Win7?

Maybe I asked too many questions on it, but in the year of 2024, when I hear "works on Win7", my impression is .net framework 4.x, not .net 6/7/8, and then back to Windows c# GUI framework, it's WPF (.net framework).

**Reply by DevTKSS** [reply]
Thats exactly what I would be thinking. @felixf4xu
Especially because Microsoft at least list Framework 4.7+ as option for lets say business application and you can still build new applications using this. Some companys still require to use Framework because they might have other applications that will not be switched to .net (with Framework)

Uno does advertise, to be running on all targets, what would mean in the common understanding also net.framework like in a wpf like you said. If its not, that statement would be false and clearly stated everywhere, especially on such ui framework like wpf, uwp and uno islands descriptions

**Reply by jeromelaban** [reply]
Windows 7 has been out of support for almost 5 years now, and the dependents from Microsoft are also out of support. This is not something we can change in any way.

Now, that being said, Microsoft is unofficially saying that they don't make it an intentional break to make .NET 8 fail on Windows 7. It will keep working until they *really* need to break compatibility. .NET 9 may still work on Windows 7 and it's the same philosophy as far as I can tell.
